wafer/schedule/views.py

```diff
@@ -203,3 +210,70 @@ def get_context_data(self, **kwargs):
                     # Must overlap with current slot
                     item['note'] = 'current'
         return context
+
+
+class ScheduleItemViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows groups to be viewed or edited.
+    """
+    queryset = ScheduleItem.objects.all()
+    serializer_class = ScheduleItemSerializer
+
+
+class ScheduleEditView(TemplateView):
+    template_name = 'wafer.schedule/edit_schedule.html'
+    date_field = 'date'
+    allow_future = True
+
+    def get_context_data(self, id=None, **kwargs):
+        context = super(ScheduleEditView, self).get_context_data(**kwargs)
+
+        days = Day.objects.all()
+
+        day = days.first()
+        if id:
+            day = days.get(id=id)
+
+        accepted_talks = Talk.objects.filter(status=ACCEPTED)
+        venues = Venue.objects.filter(days__in=[day])
+        slots = Slot.objects.filter(day=day).select_related(
+            'day', 'previous_slot').prefetch_related(
+            'scheduleitem_set', 'slot_set')
```

Comment:
day may not be set for slots that have prev_slot set, so this can miss slots of interest.

This arguably should be fixed by changing how the slot model behaves, though.